solr/solrj/src/java/org/apache/solr/client/solrj/impl/Http2SolrClient.java

```diff
@@ -505,6 +505,9 @@ private NamedList<Object> processErrorsAndResponse(
     if (contentType != null) {
       mimeType = MimeTypes.getContentTypeWithoutCharset(contentType);
       encoding = MimeTypes.getCharsetFromContentType(contentType);
+      if (parser.getWriterType().equals("json") && encoding == null) {
+        encoding = FALLBACK_CHARSET.name();
+      }
```

Comment:
interesting; what led to this change?

Reply:
Without the correct encoding specified here I got an exception when requesting the manifest.json file, since the SHA did not match the expected. 
I tried to set the content type for the response to contain the encoding but when I dug deeper it turned out the we get if from the BinaryResponseParser and I was more afraid to change it there since it would affect much more then just the manifest.json so I added this if here.

Reply:
This is very much a hack (a needless special case) so I'm trying to avoid this so that Http level code needn't have assumptions on formats when we already have abstractions for those (the ResponseParser hierarchy) where such logic might go.  Since it seems your special case is JSON specific, how is it that BinaryResponseParser (JAVABIN) is relevant?

Reply:
I can agree with you that this feels a bit hacky. Let me dig a bit deeper into it

Reply:
Seems you fixed it or something -- latest state is clean.